core/authn.go

```diff
@@ -34,26 +36,54 @@ type tokenResult struct {
 
 func (a *apiAuthn) handler(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-		err := a.auth(req)
-		if err != nil {
-			errorFormatter.Write(req.Context(), rw, err)
+		token, err0 := a.tokenAuth(req)
+		ctx := req.Context()
+		if err0 == nil && token != "" {
+			// if this request was successfully authenticated, pass the token along
+			ctx = context.WithValue(ctx, "token", token)
```

Comment:
Oh whoops, so I see the `CredentialStore` now. If I have one other suggestion it would be to separate it by credential types, and (perhaps in a separate PR) also toss the X.509 client certificate used in the request into there (if, for no other reason, so it's easy to introspect)

It's nice having all the different credential types in one place because sometimes you want to use them in combination (e.g. X.509 + token for [Token Binding](https://datatracker.ietf.org/wg/tokbind/charter/))